### PR TITLE
refactor: migrate graphql to http spec

### LINKED ISSUE #731: Migrate tests that depend on jsonplaceholder from `graphql_spec` to `http_spec`
Currently there are quite a few tests in graphql_spec that connect to jsonplaceholder while executing. These tests are not only slow, they don't pass if one is offline.

**Proposal**
- Move execution tests from `graphql_spec` to `http_spec`.
- Ensure the upstream services are mocked and allows us to run tests offline.

**Technical Requirements**
- Ensure there is no drop in coverage.
- Ensure that all scenarios are 1 on 1 migrated and no edge case is missed while migrating the tests.
- Ensure tests work offline.

### PR DESCRIPTION
**Pull request by ologbonowiwi**
**Summary:**  
I've moved the changes that made actual requests from `graphql_spec` into `http_spec`.

**Issue Reference(s):**  
Fixes #731
/claim #731

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [ ] I have performed a self-review of my own code.

### CODE REVIEW

**Comment by ologbonowiwi** [comment]
Hey @tusharmath, the down of -0.08% was on [src/http/client.rs](https://app.codecov.io/gh/tailcallhq/tailcall/pull/751/blob/src/http/client.rs), because we don't run actual calls against HTTP servers anymore. 

The only way to have it covered would be to have a test that runs with an actual HTTP client instead of making mocked requests. Please let me know what you think.
In case we need to cover it, the most straightforward way to do it would be by letting at least one file under `/graphql/passed` run actual HTTP requests.

File: tests/http/config/add-field.graphql [hunk 1]
```diff
@@ -20,8 +20,8 @@ type Query {
...
-query @expect(json: {data: {user: {lat: "-37.3159"}}}) {
-  user {
-    lat
-  }
-}
+# query @expect(json: {data: {user: {lat: "-37.3159"}}}) {
+#   user {
+#     lat
+#   }
+# }
```
Comment (tusharmath) [inline tests/http/config/add-field.graphql:27]: Can you delete commented code please

File: tests/http/config/batching-default.graphql [hunk 2]
```diff
@@ -22,11 +22,11 @@ type User {
...
-  }
-}
+# query @expect(json: {data: {posts: [{userId: 1, user: {id: 1}}, {userId: 2, user: {id: 2}}]}}) {
+#   posts {
+#     user {
+#       id
+#     }
+#     userId
+#   }
+# }
```
Comment (tusharmath) [inline tests/http/config/batching-default.graphql:32]: Can you delete commented code please

File: tests/http/config/batching-group-by-default.graphql [hunk 3]
```diff
@@ -23,12 +23,12 @@ type User {
...
-}
+# #> client-query
+# query @expect(json: {data: {posts: [{userId: 1, user: {id: 1}}, {userId: 2, user: {id: 2}}]}}) {
+#   posts {
+#     user {
+#       id
+#     }
+#     userId
+#   }
+# }
```
Comment (tusharmath) [inline tests/http/config/batching-group-by-default.graphql:34]: Can you delete commented code please

File: tests/http/config/batching-group-by.graphql [hunk 4]
```diff
@@ -23,12 +23,12 @@ type User {
...
-}
+# #> client-query
+# query @expect(json: {data: {posts: [{userId: 1, user: {id: 1}}, {userId: 2, user: {id: 2}}]}}) {
+#   posts {
+#     user {
+#       id
+#     }
+#     userId
+#   }
+# }
```
Comment (tusharmath) [inline tests/http/config/batching-group-by.graphql:34]: Can you delete commented code please

**Comment by ologbonowiwi** [comment]
done @tusharmath
